env0/resource_environment_output_configuration_variable.go

```diff
@@ -0,0 +1,319 @@
+	}
+
+	return string(b), nil
+}
+
+func deserializeEnvironmentOutputConfigurationVariableValue(valueStr string) (*EnvironmentOutputConfigurationVariableValue, error) {
+	var value EnvironmentOutputConfigurationVariableValue
+
+	if err := json.Unmarshal([]byte(valueStr), &value); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal value string: %w", err)
+	}
+
+	if value.OutputName == "" {
+		return nil, fmt.Errorf("after unmarshal 'outputName' is empty")
+	}
+
+	if value.ProjectId == "" {
+		return nil, fmt.Errorf("after unmarshal 'projectId' is empty")
+	}
+
+	if value.EnvironmentId == "" && value.SubEnvironmentAlias == "" {
+		return nil, fmt.Errorf("after unmarshal both 'environmentId' and 'subEnvironmentAlias' are empty")
+	}
+
+	return &value, nil
+}
+
+func getEnvironmentOutputConfigurationVariableParamsFromVariable(d *schema.ResourceData, variable *client.ConfigurationVariable) (*EnvironmentOutputConfigurationVariableParams, error) {
+	var params EnvironmentOutputConfigurationVariableParams
+	if err := readResourceData(&params, d); err != nil {
+		return nil, fmt.Errorf("schema resource data deserialization failed: %v", err)
+	}
+
+	params.Name = variable.Name
+	params.Description = variable.Description
+
+	if variable.IsReadOnly != nil {
+		params.IsReadOnly = *variable.IsReadOnly
+	} else {
+		params.IsReadOnly = false
+	}
+
+	if variable.IsRequired != nil {
+		params.IsRequired = *variable.IsRequired
+	} else {
+		params.IsRequired = false
+	}
+
+	if variable.Type == nil || *variable.Type == client.ConfigurationVariableTypeEnvironment {
+		params.Type = "environment"
+	} else {
+		params.Type = "terraform"
+	}
+
+	params.ScopeId = variable.ScopeId
+
+	switch scope := variable.Scope; scope {
+	case client.ScopeEnvironment, client.ScopeDeployment, client.ScopeWorkflow, client.ScopeProject:
+		params.Scope = string(scope)
+	default:
+		return nil, fmt.Errorf("invalid scope %s", scope)
+	}
+
+	value, err := deserializeEnvironmentOutputConfigurationVariableValue(variable.Value)
+	if err != nil {
+		return nil, fmt.Errorf("invalid value: %w", err)
+	}
+
+	params.OutputEnvironmentId = value.EnvironmentId
+	params.outputEnvironmentProjectId = value.ProjectId
+	params.OutputSubEnvironmentAlias = value.SubEnvironmentAlias
+	params.OutputName = value.OutputName
+
+	return &params, nil
+}
+
+func getEnvironmentOutputCreateParams(d *schema.ResourceData, apiClient client.ApiClientInterface) (*client.ConfigurationVariableCreateParams, error) {
+	var params EnvironmentOutputConfigurationVariableParams
+	if err := readResourceData(&params, d); err != nil {
+		return nil, fmt.Errorf("schema resource data deserialization failed: %v", err)
+	}
+
+	if params.Scope != string(client.ScopeProject) && params.IsReadOnly {
+		return nil, errors.New("'is_read_only' can only be set to 'true' for the 'PROJECT' scope")
+	}
+
+	environment, err := apiClient.Environment(params.OutputEnvironmentId)
```

Comment:
getting the environment details in order to get the environment project id (this is passed in the "Value".